File: backend-api/src/main/java/vn/aptech/backendapi/entities/Department.java
package vn.aptech.backendapi.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "departments", uniqueConstraints = {
    @UniqueConstraint(columnNames = "name")
})
public class Department extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    
    @Column(unique = true) // Optional: You can also annotate with @Column(unique = true)
    private String name;
    
    private String icon;
    private String url;
    private boolean status;
}

File: mobile/lib/models/department.dart
class Department{
   final int id;
   final String name;
   final String icon;
   final String url;
   final bool status;
   Department({required this.id, required this.name, required this.icon, required this.url, required this.status});

   factory Department.fromJson(Map<String, dynamic> json) {
     return Department(
       id: json['id'],
       name: json['name'],
       icon: json['icon'],
       url: json['url'],
       status: json['status'] as bool,
     );
   }

   Map<String, dynamic> toJson() {
     return {
       'id': id,
       'name': name,
       'icon': icon,
       'url': url,
       'status': status,
     };
   }

}